src/Posts.js

```diff
@@ -0,0 +1,245 @@
+import React,{useEffect ,useState, useRef} from 'react'
+import { useNavigate ,useLocation,useSearchParams } from 'react-router-dom';
+import CustomModal from './Modal';
+import Edit from './Edit';
+export default function Posts(props)
+ {
+  const [posts,setPosts]=useState([])
+  const [postid,setpostid]=useState(0)
+  const title=useRef()
+  const des=useRef()
+  const postarr=[]
+  const [currentPage, setCurrentPage] = useState(1);
+  let path=useNavigate()
+  const [searchparams]=useSearchParams()
+  useEffect(() => {
+    fetchPosts();
+  }, [currentPage]);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isModalOpenedit, setIsModalOpenedit] = useState(false);
+
+  let objposts=
+  {
+       
+  }
+
+  let localkeys=Object.keys(localStorage)
+    let convertedkeys=[]
+    for(let a =0;a<localkeys.length;a++)
+    {
+      if(localkeys[a].includes('@'))
+      {
+
+      }
+      else{
+        convertedkeys.push(parseInt(localkeys[a]))
+      }
+    }
+    let sortedarray=convertedkeys.sort((a, b) => b - a);
+    for(let a of sortedarray)
+    {
+      objposts[a]={}
+      
+      objposts[a]=JSON.parse(localStorage.getItem(a))
+    }
+    console.log("meri posts",objposts)
+  const fetchPosts = async () => {
+    try {
+      const response = await fetch(`https://jsonplaceholder.typicode.com/posts?_page=${currentPage}&_limit=7`);
+      if (!response.ok) {
+        throw new Error('Network response was not ok.');
+      }
+      const jsonData = await response.json();
+      setPosts(jsonData);
+    } catch (error) {
+      console.error('Error fetching data:', error);
+    }
+  };
```

Comment:
move this api call to a separate file. All api's call should be in one separate file and import those methods where ever you need.